lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -352,30 +358,45 @@ def add_line_items(xml, level_3_data)
         }
       end
 
-      def add_stored_credential(xml, options)
+      def add_stored_credential(xml, payment_method, options)
+        return unless options[:stored_credential]
         network_transaction_id = options.dig(:stored_credential, :network_transaction_id)
         case
-        when network_transaction_id.nil?
-          return
         when network_transaction_id.to_s.include?('|')
           oar_data, ps2000_data = options[:stored_credential][:network_transaction_id].split('|')
-          xml.ssl_oar_data oar_data unless oar_data.nil? || oar_data.empty?
-          xml.ssl_ps2000_data ps2000_data unless ps2000_data.nil? || ps2000_data.empty?
+          xml.ssl_oar_data oar_data unless oar_data.nil? || oar_data.empty? || payment_method.is_a?(String)
+          xml.ssl_ps2000_data ps2000_data unless ps2000_data.nil? || ps2000_data.empty? || payment_method.is_a?(String)
         when network_transaction_id.to_s.length > 22
-          xml.ssl_oar_data options.dig(:stored_credential, :network_transaction_id)
-        else
-          xml.ssl_ps2000_data options.dig(:stored_credential, :network_transaction_id)
+          xml.ssl_oar_data network_transaction_id
+        else !network_transaction_id.blank?
+          xml.ssl_ps2000_data network_transaction_id
         end
+        xml.ssl_recurring_flag recurring_flag(options) if recurring_flag(options)
+        xml.ssl_approval_code options[:approval_code] if options[:approval_code] && !payment_method.is_a?(String)
+        xml.ssl_par_value options[:par_value] if options[:par_value]
+        xml.ssl_association_token_data options[:association_token_data] if options[:association_token_data]
```

Comment:
Two new values that may be present on an initial transaction that should be passed in subsequent COF transactions.